mytardis_rocrate_builder/rocrate_builder.py

```diff
@@ -154,47 +209,86 @@ def __add_person_to_crate(self, person: Person) -> ROPerson:
         Args:
             person (Person): the person to add to the crate
         """
-        orcid_regex = re.compile(
-            r"https://orcid\.org/[0-9]{4}-[0-9]{4}-[0-9]{4}-[0-9]{3}[0-9X]{1}"
-        )
+
+        # orcid_regex = re.compile(
+        #     r"https://orcid\.org/[0-9]{4}-[0-9]{4}-[0-9]{4}-[0-9]{3}[0-9X]{1}"
+        # )
         upi_regex = re.compile(r"^[a-z]{2,4}[0-9]{3}$")
-        person_id = None
-        # Check to see if any of the identifiers are orcids - these are preferred
-        for identifier in person.identifiers:
-            if _ := orcid_regex.fullmatch(identifier):
-                person_id = identifier
+        person_id = person.id
+        # # Check to see if any of the mt_identifiers are orcids - these are preferred
+        # for identifier in person.mt_identifiers:
+        #     if _ := orcid_regex.fullmatch(identifier):
+        #         person_id = identifier
         # If no orcid is found check for UPIs
-        if not person_id:
-            for identifier in person.identifiers:
+        if not upi_regex.fullmatch(str(person_id)):
+            for identifier in person.mt_identifiers:
                 if _ := upi_regex.fullmatch(identifier):
                     person_id = identifier
         # Finally, if no orcid or UPI is found, use the first identifier
-        if not person_id:
-            person_id = person.identifiers[0]
+        # if not person_id:
+        #     person_id = person.mt_identifiers[0]
+        person_obj = ROPerson(
+            self.crate,
+            person_id,
+            properties={"name": person.name, "email": person.email},
+        )
+
         if not any(
             (
                 entity.type in ["Organization", "ResearchOrganization"]
                 and entity.id == person.affiliation.id
             )
             for entity in self.crate.get_entities()
         ):
-            self.__add_organisation(person.affiliation)
-        person_obj = ROPerson(
-            self.crate,
-            person_id,
-            properties={
-                "name": person.name,
-                "email": person.email,
-                "affiliation": person.affiliation.id,
-            },
-        )
-        if len(person.identifiers) > 1:
-            for identifier in person.identifiers:
+            person_obj.append_to(
+                "affiliation", self.__add_organisation(person.affiliation)
+            )
+        if len(person.mt_identifiers) > 1:
```

Comment:
See comment in the organisation function. We need to account for a single identifier